parsec/service_nursery.py

```diff
@@ -191,5 +194,42 @@ def wrap_started(value: object = None) -> None:
             child_task_scopes.shield = False
 
 
+def collapse_multierror(multierror):
+    exceptions = multierror.exceptions
+    pick = exceptions[0]
+    try:
+        name = f"{type(pick).__name__}AndFriends"
+        cls = type(name, (type(pick), trio.MultiError), {})
+        result = super(cls, cls).__new__(cls)
+        result.__dict__.update(pick.__dict__)
+        result.args = pick.args
+        result.exceptions = exceptions
+        result.__cause__ = multierror.__cause__
+        result.__context__ = multierror.__context__
+        result.__traceback__ = multierror.__traceback__
+        result.__suppress_context__ = True
+        return result
+    except Exception:
+        logger.exception("Cound not create a collapsed exception")
```

Comment:
the logger should display the multierror given otherwise it is lost (and we really want it to know why the unexpected exception occurred !)